onramps.php

```diff
@@ -0,0 +1,136 @@
+add_action('activated_plugin', 'store_brand_plugin_name_on_activation', 10, 1);
+add_action('deactivated_plugin', 'store_brand_plugin_name_on_deactivation', 10, 1);
+add_action('admin_init', 'check_and_install_plugin');
+
+/**
+ * Remove the delete button for brand plugin
+ */
+function remove_delete_button($actions, $plugin_file) {
+    $BH_plugin = 'bluehost-wordpress-plugin/bluehost-wordpress-plugin.php';
+    $HG_plugin = 'wp-plugin-hostgator/wp-plugin-hostgator.php';
+
+    if ($plugin_file === $BH_plugin || $plugin_file === $HG_plugin) {
+        unset($actions['delete']);
+    }
+
+    return $actions;
+}
+
+
+/**
+ * Add placeholder for the brand plugin if it is not there in the plugin list 
+ */
+function modify_plugins_list ($plugins) {
+    $add_plugin_placeholder = true;
+
+    $brand_plugin = get_option( 'nfd_brand_plugin' );
+
+    if (array_key_exists($brand_plugin, $plugins)) {
+        $add_plugin_placeholder = false;
+    }
+
+    if($add_plugin_placeholder){
+        if("The Bluehost Plugin " === $brand_plugin) {
+            $plugins['bluehost-wordpress-plugin/bluehost-wordpress-plugin.php'] = array(
+                "Name" => "The Bluehost Plugin",
+                "TextDomain"=> "",
+                "Author"    => "Bluehost",
+                "Version" => "",
+                "PluginURI" => "",
+                "AuthorURI" => "",
+                "Description" => ""
+            );
+        }
+        else if("The Hostgator Plugin " === $brand_plugin) {
+            $plugins['wp-plugin-hostgator/wp-plugin-hostgator.php'] = array(
+                "Name" => "The Hostgator Plugin",
+                "TextDomain"=> "",
+                "Author"    => "Hostgator",
+                "Version" => "",
+                "PluginURI" => "",
+                "AuthorURI" => "",
+                "Description" => ""
+            );
+        }
+       
+    }
+    return $plugins;
+}
+
+/**
+ * save brand plugin name on activation
+ */
+function store_brand_plugin_name_on_activation($plugin) {
+    update_brand_plugin_name($plugin);
+}
+
+/**
+ * save brand plugin name on deactivation
+ */
+function store_brand_plugin_name_on_deactivation($plugin) {
+    update_brand_plugin_name($plugin);
+}
+
+function update_brand_plugin_name($plugin) {
+    $current_brand_plugin = get_option('nfd_brand_plugin', '');
+
+    $plugin_data = get_plugin_data(WP_PLUGIN_DIR . '/' . $plugin);
+    $plugin_name = $plugin_data['Name'];
+
+    if ('The Bluehost Plugin' === $plugin_name || 'The Hostgator Plugin' === $plugin_name) {
+        if (empty($current_brand_plugin) || $current_brand_plugin !== $plugin_name) {
+            update_option('nfd_brand_plugin', $plugin_name);
+        }
+    }
+}
+
+/**
+ * On click of activate install and activate the plugin
+ */
+function check_and_install_plugin() {
+    if (isset($_GET['action']) && $_GET['action'] == 'activate' && isset($_GET['plugin']) && $_GET['plugin'] == 'wp-plugin-hostgator/wp-plugin-hostgator.php') {
```

Comment:
Does the `nfd_brand_plugin` option get set during a "fake" activation?

Reply:
The brand plugin will be stored in the database when it is deactivated. ( While activating, it will update the database if not set in the database.)